### DISCUSSION TITLE
Difference between ContentControl and UserControl?

### DISCUSSION THREAD

**Discussion by martinrhan** [AvaloniaUI/Avalonia#10162] [Q&A]
I looked though documents and did some Google search but cannot get the answer. The document say UserControl is better for View, but how is it so?

**Comment by timunie** [comment]
You can see UserControl as a kind of "Page". You can create fragments of your view inside and place it into your Winow or what ever. A `ContentControl` is more or less your base class if you want to create a new, reusable control for a control library. 

Hope that makes some sense.

**Comment by lukaszossowski** [comment]
I'd also add that content control can be used to display user controls. In my apps with side menus I have the main window with side menus populated and responsible for navigation, connected to MainWindowViewModel which does the navigation by changing the currently active page viewmodel and the pages themselves are displayed in a content control like that:

```
<ContentControl Grid.Row="2" Content="{Binding CurrentPageVM}" >
		<ContentControl.DataTemplates>
			<DataTemplate DataType="backend:ComponentPageViewModel">
				<local:ComponentPage />
			</DataTemplate>
		</ContentControl.DataTemplates>
		<ContentControl.DataTemplates>
			<DataTemplate DataType="backend:ProductPageViewModel">
				<local:ProductPage />
			</DataTemplate>
		</ContentControl.DataTemplates>
		<ContentControl.DataTemplates>
			<DataTemplate DataType="backend:SalePageViewModel" >
				<local:SalesPage />
			</DataTemplate>
		</ContentControl.DataTemplates>
</ContentControl>
```
so it has content bound to a pagevm and displays appropriate usercontrol (page). 

User controls can also be used as smaller parts of GUI, for example if you have your own custom colour picker that consists of multiple controls you can place them inside a usercontrol and reuse it in different parts of your app.

**Reply by a70n3** [reply]
Glad I found your comment, I've used Window.Resources for the datatemplate just like what I've been doing in WPF, but it only shows the Class Name.
